Infrastructure/Resources/New-ResourceGroup.ps1

```diff
@@ -12,36 +12,114 @@ The location of the Resource Group. This is limited to West Europe and North Eur
     [String]$Location,
-    [Parameter(Mandatory = $true)]
+    [Parameter(Mandatory = $True)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Name,
+    [Parameter(Mandatory = $False)]
     [ValidateNotNullOrEmpty()]
-    [String]$Name
+    [String]$TagConfigPath
 )
 
 # --- Import Azure Helpers
 Import-Module (Resolve-Path -Path $PSScriptRoot\..\Modules\Azure.psm1).Path
 Import-Module (Resolve-Path -Path $PSScriptRoot\..\Modules\Helpers.psm1).Path
 
-Write-Log -LogLevel Information -Message "Checking for existing Resource Group $Name"			 
+Write-Log -LogLevel Information -Message "Checking for existing Resource Group $Name"
 $ExistingResourceGroup = Get-AzureRmResourceGroup -Name $Name -ErrorAction SilentlyContinue
 
+# --- Enumerate Tag Value's from Config File if Parameter is set 
+If ($TagConfigPath) {
+    $TagConfig = Get-Content -Path $TagConfigPath -Raw | ConvertFrom-Json
+    $ConfigTagValue = $TagConfig.Value | ConvertTo-Json
+    $ConfigTagName = $TagConfig.Name 
+    # --- Define & Set variables for Tags
+    $TagPresentValueCorrect = $True 
+    $TagValueUpdated = $False
+    $Tags = $ExistingResourceGroup.Tags
+}
+
+# --- Create Resource Group if it doesn't exist
 if (!$ExistingResourceGroup) {
     try {
         Write-Log -LogLevel Information -Message "Creating Resource Group"
-        $null = New-AzureRmResourceGroup -Location $Location -Name $Name
+        If (!$TagConfigPath) {
+            $null = New-AzureRmResourceGroup -Location $Location -Name $Name
+        }
+        Else {
+            $null = New-AzureRmResourceGroup -Location $Location -Name $Name -Tag @{$TagConfig.Name = $ConfigTagValue}
+        }
     }
     catch {
         throw "Could not create Resource Group $Name : $_"
     }
 }
 
-Write-Output ("##vso[task.setvariable variable=ResourceGroup;]$Name")
-Write-Output ("##vso[task.setvariable variable=Location;]$Location")
\ No newline at end of file
+# --- \\Tags Section\\
+# --- If resource already exists check the Tags are present and hold the same values as the Config File       
+if ($ExistingResourceGroup -and $TagConfigPath) {
+    try {
+        Write-Log -LogLevel Information "Resource Exists - Checking for Valid Tags"
+        # --- Enumerate all tags
+        If ($Tags -ne $null) {
+                foreach ($Tag in $Tags.GetEnumerator()) {
+                    # --- See if the tag name and value match the Config values
+                    If ($($Tag.Name) -eq $TagConfig.Name -and $($Tag.Value) -ne $ConfigTagValue) {    
+                            $TagPresentValueCorrect = $False    
+                    }
+                    If ($($Tag.Name) -eq $TagConfig.Name -and $($Tag.Value) -eq $ConfigTagValue) {    
+                            $TagValueUpdated = $True    
+                    }
+                    
+                }
+            }
+            # --- Once all tags enumerated 
+            # --- Overwrite value if Tag already present
+            If (!$TagPresentValueCorrect) {
```

Comment:
If this value is flipped to be $TagPresentValueIncorrect here and in line 96 then there will be no need to define the variables mentioned in my previous comment.

Reply:
Not sure I understand ?